slick-hikaricp/src/main/scala/slick/jdbc/hikaricp/HikariCPJdbcDataSource.scala

```diff
@@ -58,12 +58,6 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
 
     // Infrequently used
 
-    // `initializationFailFast` is deprecated and should be replaced by
-    // `initializationFailTimeout`. See HikariCP docs for more information:
-    // https://github.com/brettwooldridge/HikariCP#infrequently-used
-    // But we still respect the value if it configured.
-    c.getBooleanOpt("initializationFailFast").foreach(hconf.setInitializationFailFast)
-
```

Comment:
@matthiassig12 As noted in the [change log](https://github.com/brettwooldridge/HikariCP/blob/dev/CHANGES) for HikariCP, methods that were previously marked as deprecated in v2.x were removed in v3.0.

I would suggest that simply removing this code is probably *not* the correct fix.  Support for ``initializationFailTimeout`` should be added.  Additionally, if you wish to maintain backward compatibility, then when ``initializationFailFast`` is set to ``true`` you can set ``initializationFailTimeout=1`` and when ``initializationFailFast`` is set to ``false`` you can set ``initializationFailTimeout=0``.

Reply:
It appears that initializationFailTimeout support has already been added (https://github.com/slick/slick/blob/master/slick-hikaricp/src/main/scala/slick/jdbc/hikaricp/HikariCPJdbcDataSource.scala#L71).

The backwards compatiblity thing is a very good point, I'll add that.

Reply:
I have updated the PR with your suggestions, but I'm not quite sure about the defaults given the already defined value of -1 in this file, I'm wondering if -1 also makes sense for when failFast is set to false?

Reply:
@matthiassig12 merging this unless there are any comments. I'm not sure if this affect backward compatibility. 

CC: @d6y @brettwooldridge

Reply:
The comment here explains why the value was `-1` if the configuration is not present:

https://github.com/slick/slick/blob/6c7027eba3f5d7a31e4e6ef97c831fbdefa648f8/slick-hikaricp/src/main/scala/slick/jdbc/hikaricp/HikariCPJdbcDataSource.scala#L67-L70

I think that maintaining this value won't affect backward compatibility in any way.